File: services/nuldoc/nuldoc-src/components/GlobalFooter.ts
import { Config } from "../config.ts";
import { elem, Element } from "../dom.ts";

export default function GlobalFooter({ config }: { config: Config }): Element {
  return elem(
    "footer",
    { class: "footer" },
    `&copy; ${config.site.copyrightYear} ${config.site.author}`,
  );
}

File: services/nuldoc/lib/nuldoc/components/global_footer.rb
module Nuldoc
  module Components
    class GlobalFooter < DOM::HTMLBuilder
      def initialize(config:)
        super()
        @config = config
      end

      def build
        site = @config.site
        footer(class: 'footer') { text "&copy; #{site.copyright_year}-#{site.current_year} #{site.author}" }
      end
    end
  end
end
